feat(axe-core-4.9.1): update to use axe-core 4.9.1

rohittarpara opened the pull request:
#### Details

This PR upgrades axe-core to 4.9.1 according to the instructions in CONTRIBUTING.md.

##### Motivation

feature work 🚀

##### Context

<!-- Are there any parts that you've intentionally left out-of-scope for a later PR to handle? -->

<!-- Were there any alternative approaches you considered? What tradeoffs did you consider? -->

#### Pull request checklist
<!-- If a checklist item is not applicable to this change, write "n/a" in the checkbox -->
- [x] PR title respects [Conventional Commits](https://www.conventionalcommits.org) (starts with `fix:`, `feat:`, etc, and is suitable for user-facing release notes)
- [x] PR contains no breaking changes, **OR** description of both PR **and final merge commit** starts with `BREAKING CHANGE:`
- [x] Verify PR title and final merge commit contain space after `:` for example `feat(feature): feature title` otherwise it will not be considered by semantic-release for release.
- [ ] (if applicable) Addresses issue: #0000
- [ ] Added relevant unit tests for your changes
- [x] Ran `yarn precheckin`
- [x] Verified code coverage for the changes made

madalynrose reviewed (changes requested):
Per the instructions in `CONTRIBUTING.md`, I think we also need to update the `devDependencies` in `package.json` for `@axe-core/cli` and `@axe-core/puppeteer`. And then regenerate the resources. Additionally, `src/cli.test.ts` needs to be updated similar to how `src/index.test.ts` was updated.

rohittarpara commented:
@madalynrose i updated version for @axe-core/cli and @axe-core/puppeteer. 
For the CONTRIBUTING.md i changed the version. Is there any other thing i need to update in CONTRIBUTING.md?

madalynrose commented on package.json:
```diff
@@ -22,8 +22,8 @@
     "devDependencies": {
-        "@axe-core/cli": "^4.8.0",
-        "@axe-core/puppeteer": "^4.8.0",
+        "@axe-core/cli": "^4.9.1",
+        "@axe-core/puppeteer": "^4.9.1",
```
These are usually specified using the minor version number, with patch version at 0
```suggestion
        "@axe-core/cli": "^4.9.0",
        "@axe-core/puppeteer": "^4.9.0",
```

madalynrose reviewed (changes requested):
one small adjustment but this looks good!

v-viyada commented:
/azp run

ada-cat commented:
:tada: This PR is included in version 2.12.0 :tada:

The release is available on:
- [npm package (@latest dist-tag)](https://www.npmjs.com/package/axe-sarif-converter/v/2.12.0)
- [GitHub release](https://github.com/microsoft/axe-sarif-converter/releases/tag/v2.12.0)

Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot :package::rocket:
